consensus/consensus_service.go

```diff
@@ -77,24 +78,30 @@ func (consensus *Consensus) UpdatePublicKeys(pubKeys, allowlist []bls_cosi.Publi
 	// TODO: use mutex for updating public keys pointer. No need to lock on all these logic.
 	consensus.pubKeyLock.Lock()
 	consensus.Decider.UpdateParticipants(pubKeys, allowlist)
-	consensus.getLogger().Info().Msg("My Committee updated")
-	for i := range pubKeys {
-		consensus.getLogger().Info().
-			Int("index", i).
-			Str("BLSPubKey", pubKeys[i].Bytes.Hex()).
-			Msg("Member")
-	}
-
 	allKeys := consensus.Decider.Participants()
+	consensus.pubKeyLock.Unlock()
 	if len(allKeys) != 0 {
-		consensus.LeaderPubKey = &allKeys[0]
+		first := consensus.Decider.FirstParticipant(
```

Comment:
why do such change? and deviate from simply getting the first key in the committee list and the first leader?

Reply:
rolled back